src/Extensions/Extension.php

```diff
@@ -128,6 +117,16 @@ public function isEnabled()
     }
 
     /**
+     * Loads the icon information from the composer.json.
+     *
+     * @return array|null
+     */
+    public function getIcon()
+    {
+        return $this->installedExtension->getIcon();
```

Comment:
What happens when the extension is not installed ?

By the way, it's how I planned to do it :) This should also be put in the API so we can display it for uninstalled applications